CityOfDetroit/bloom
# feat: update major dependencies

This PR addresses #1591 


## Description

This upgrades several dependencies that have already been upgraded in core this includes:
- Node from 14 to 18
- Next to 12
- React to 18
- Cypress from 9 to 12

It also accomplishes the following:
- removes storybook from detroit-ui-components
- update all test ids to data-testid

## How Can This Be Tested/Reviewed?

This touches every page so it is recommended to go to every page and do a quick check. These changes are already in and have been tested so we don't have to be as thorough, but there are some changes that are unique so a quick passover is recommended. 

You will need to test locally since it has to have a new deployed version of the backend

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
